### ISSUE TITLE
LabeledExpression false positive when tagging return with scope

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5132]
## Steps to Reproduce
```kotlin
interface Context {
	fun <T> read(name: String): T?
	fun Context.get(route: String, body: Receiver.() -> Unit)
}

interface Receiver {
	fun respond(message: String)
}

fun Context.register() {
	get("/abc") {
		val data = read<String>("something")
			?: return@get
		respond("got $data")
	}
}

```
## Expected Behavior
No detection

## Observed Behavior
> LabeledExpression - [Expression with labels increase complexity and affect maintainability.]

at `return@get`

## Context
ktor and any other similar API where I have to say what to return from otherwise, I can't write the logic I need.

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.5
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.cinema/blob/1a45537b0b06b69816d4ede8bd7f2e675fb4efc9/Heroku/backend/endpoint/src/main/kotlin/net/twisterrob/cinema/cineworld/backend/endpoint/film/FilmsController.kt#L41-L45

**Comment by BraisGabin** [comment]
This is not a false positive. "LabeledExpressions" is not "UnnecessaryLabeledExpressions". This rule flags ALL the labeler expressions. From its documentation:

> This rule reports labeled expressions. Expressions with labels generally increase complexity and worsen the maintainability of the code. Refactor the violating code to not use labels instead. Labeled expressions referencing an outer class with a label from an inner class are allowed, because there is no way to get the instance of an outer class from an inner class in Kotlin.

(Please reopen if I missed something)

**Comment by TWiStErRob** [comment]
Fair enough. I misunderstood the intent of the rule.
